### PR TITLE
Started Working on the CRUD - Fetching Data from Prisma

### PR DESCRIPTION
**Pull request by denimozh**
Started Working on the CRUD - Fetching Data from Prisma. I haven't finished adding all the products into the database. Finished the Fetching Data function for the Catalog. Started working on the Fetching Data function for products.

### CODE REVIEW

File: src/app/api/products/route.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            where:{
+                ...(cat ? {catSlug:cat} : {isFeatured:true}),
+            }
+        });
+
+        return new NextResponse(JSON.stringify(products), {status:200});
+
+    } catch (error) {
+        console.log(error);
+        return new NextResponse(JSON.stringify({message:"Something went wrong!"}), {status:500});
```
Comment (AliakseiDeboi) [inline src/app/api/products/route.ts:21]: You can create a separate enum for storing error messages, the other one for statutes. It'd be useful, I think

File: src/utils/connect.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { PrismaClient } from "@prisma/client";
+
+const globalForPrisma = globalThis as unknown as {
+    prisma: PrismaClient | undefined
+}
+
+export const prisma = globalForPrisma.prisma ?? new PrismaClient()
```
Comment (AliakseiDeboi) [inline src/utils/connect.ts:7]: ;
and in some other places
we will use ESLINT later to fix it, no worries

**Review by AliakseiDeboi — APPROVED** [review]
Looks good! There are some comments, but you sill can merge it

File: src/app/api/products/route.ts (hunk 1 above)
Comment (denimozh) [reply src/app/api/products/route.ts:21]: Oh okay will do

File: src/utils/connect.ts (hunk 2 above)
Comment (denimozh) [reply src/utils/connect.ts:7]: Okay

**Pull request merged**